internal/controller/sync.go

```diff
@@ -169,10 +169,10 @@ func (r *Route) hasValidCertificate(route *routev1.Route) bool {
 			)
 		}
 	}
-	// As there is no renew-before, is the cert less than 2/3 through its life?
+	// As there is no renew-before, is the cert more than 2/3 through its life?
 	totalDuration := cert.NotAfter.Sub(cert.NotBefore)
 	timeToExpiry := cert.NotAfter.Sub(time.Now())
-	if timeToExpiry < (totalDuration * 2 / 3) {
+	if timeToExpiry < (totalDuration * 1 / 3) {
 		r.eventRecorder.Event(route, corev1.EventTypeNormal, ReasonIssuing, "Issuing cert as the existing cert is more than 2/3 through its validity period")
 		return false
 	}
```

Comment:
The old code looks OK. As I understand it, the old code was doing this:

"As there is no renew-before annotation, we want to know if the certificate is still valid. If the duration until the certificate expires isn't beyond ⅔ of the total certificate validity duration, we can return `false` (meaning that the certificate is still valid).

Reply:
What do you think @seandilda?

Reply:
Sorry for the delay on the response.

I believe the `timeToExpiry` is how much of the cert life is less.  This means the old code is saying "If the time remaining is less than 2/3 of the cert life (if we've gone at least 30 days into a 90 day cert), then we consider the cert needing to be renewed and return `false`."

Am I misreading that? 

All of my certs are 90 days certs and they're being renewed every 30 days, which lines up with my reading of the code.

Reply:
A diagram might help. In the following diagram, the certificate doesn't need to be renewed.

```text
                                                  timeToExpiry
                                       <------------------------------->
                                                   totalDuration*2/3
                                              <------------------------>
+--------------------------------------+------+------------------------+
notBefore                             now     renewalTime       notAfter
t=0                                  t=56     t=60                  t=90
```

By reading the diagram, we can visually see that "timeToExpiry" is bigger than "totalDuration*2/3". In other words, we have:

```
timeToExpiry >= totalDuration*2/3
```

Now, looking at the formula, `timeToExpiry < totalDuration*2/3` is `false` which is exactly what we wanted.

Does it make sense?

Reply:
Thank you for the diagram.  It is very helpful.  However, I'm not sure `totalDuration*2/3` is drawn properly.
In this example, I believe `totalDuration` is `90`.  Two thirds of `90` is `60`.  However, it is draw as it if it is `30` in this diagram.

Let's look at `t=31`.  In this case `timeToExpiry` will be `59`.   `timeToExpiry < totalDuration*2/3` will be `true`, causing the cert to be renewed early.

Reply:
You are totally correct, my diagram is incorrect!! I cannot believe I wasn't able to calculate 90*2/3 😅

Here is the correct diagram:

```text
                                                  timeToExpiry
                                       <------------------------------->
                                             totalDuration*2/3
                     <------------------------------------------------->
+--------------------+-----------------+-------------------------------+
notBefore                             now                       notAfter
t=0                t=30               t=56                          t=90
```

I now agree with you: `timeToExpiry < totalDuration*2/3` is incorrect! We want to renew when timeToExpiry reaches 1/3 of the totalDuration, not 2/3!

With the formula `timeToExpiry < totalDuration*1/3`, the diagram makes more sense:

```text
                                                  timeToExpiry
                                       <------------------------------->
                                                   totalDuration*⅓
                                              <------------------------>
+--------------------------------------+------+------------------------+
notBefore                             now     renewalTime       notAfter
t=0                                  t=56     t=60                  t=90
```

That's definitely a bug, excellent work @seandilda!

Reply:
The variable names are confusing me, and I think it also led to the original mistake.

```go
age := now - notBefore
expiryThreshold := totalDuration * 2 / 3
if age > expiryThreshold {
    // Renew.
}
```

The words "age" and "expiryThreshold" are things that I can easily grasp in my head, unlike "timeToExpiry". The diagram starts from the left instead of the right:

```
                  age                             totalDuration*1/3
+-------------------------------------->
               expiryThreshold
+--------------------------------------------->
+--------------------------------------+------+------------------------+
notBefore                             now     renewalTime       notAfter
t=0                                  t=56     t=60                  t=90
```

Here, the age is lower than the threshold, so we don't renew.